Changelog.md

```diff
@@ -24,6 +24,7 @@ Bugfixes:
  * SMTChecker: Fix false negative when a verification target can be violated only by trusted external call from another public function.
  * SMTChecker: Fix internal error caused by using external identifier to encode member access to functions that take an internal function as a parameter.
  * Standard JSON Interface: Fix an incomplete AST being returned when analysis is interrupted by certain kinds of fatal errors.
+ * TypeChecker: Function types other than ``internal`` and ``external`` cannot be used anymore in complex expressions.
```

Comment:
No, this is **not** an important bug.

Reply:
So, the bug remains where it's at - but what about the phrasing @cameel?

Reply:
Actually, looking at it now, maybe the original was not that bad. I just realized that the bit about implicit conversions in my suggestion was covered in #14365 rather than here so that's out of place.

Other than that the original is just imprecise: it refers to kinds, rather than visibility (so "external" here does not include external library functions, which I bet will not be obvious to people) and also does not mention `Kind::Declaration`.

Personally, I'd go with something like this:

```
 * TypeChecker: Disallow using certain unassignable function types in complex expressions.
```

or maybe even this, though this may be going too far by making the change look simpler than it really is:
```
 * TypeChecker: Restrict the use of function types in complex expressions to types that can also be used to declare variables.
```


Both are just as imprecise, but at least relate the effects of this change to things that people can easily observe. I.e. the fact that the types we're disallowing in are mostly not nameable and you cannot assign values of such types or return them from functions.

Reply:
First one it is.